zerver/lib/streams.py

```diff
@@ -172,9 +172,12 @@ def send_stream_creation_event(
 
 
 def check_channel_creation_permissions(
-    user_profile: UserProfile, stream_dict: StreamDict, is_default_stream: bool
+    user_profile: UserProfile,
+    is_default_stream: bool,
```

Comment:
I think this change can be in the preparatory commit which adds this function. And also we should make the arguments after `user_profile` as keyword-only for better readability.